src/components/input/select/index.tsx

```diff
@@ -168,8 +176,19 @@ export class SmoothlyInputSelect implements Input, Editable, Clearable, Componen
 		}
 	}
 	handleShowOptions(event?: Event): void {
+		event && event.stopPropagation()
+		const wasButtonClicked =
+			event &&
+			event
+				.composedPath()
+				.some(
+					element =>
+						"tagName" in element && typeof element.tagName === "string" && this.buttonTagNames.includes(element.tagName)
+				)
+
```

Comment:
If we grab the ref for the `<div>` and the `<smoothly-icon>` (this really should be a smoothly button with an icon inside of it. change it if you want or skip for now) to varaibles
```ts
private buttons?: HTMLElement
private toggle?: HTMLElement
``` 
we should be able to write this as 
```ts
event?.composedPath().includes(this.buttons) && !event.composedPath().includes(this.toggle)
```
Which i think is cleaner and we don't have to use those hard coded tag names. this will handle any slotted content you might throw at it with no changes required.